### PR TITLE
Support for field, property and enum constant references

### PR DESCRIPTION
**Pull request by kjonescertinia**
Needs some manual testing to check for edge cases.

### CODE REVIEW

File: jvm/src/main/scala/com/nawforce/apexlink/cst/Expressions.scala [hunk 1]
```diff
@@ -443,6 +443,10 @@ final case class DotExpressionWithId(expression: Expression, safeNavigation: Boo
       DotExpression.findField(name, inputType, context.module, input.isStatic)
     if (field.nonEmpty) {
       context.addDependency(field.get)
+      field.get match {
+        case ref: Referenceable => ref.addReferencingLocation(location)
+        case _                  =>
+      }
```
Comment (pwrightcertinia) [inline jvm/src/main/scala/com/nawforce/apexlink/cst/Expressions.scala:449]: Feels not great having to sprinkle these checks in. Similar to checking for `ApexMethodLike` in other places I assume.

You could wrap it in a type assertion func or something implicit for apex types that contains the match but at best it would be `field.whenApex(_.addReferencingLocation(location))`

**Review by pwrightcertinia — APPROVED** [review]
